### ISSUE TITLE
Warn if uploading configuration will overwrite someone elses changes

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#4923]
Technical users use medic-conf to make configuration changes to instances and hopefully remember to commit the changes to git to track, back up, and share their work.

This can overwrite other configuration changes if...

1. a user has made changes through the admin app,
2. a user has made changes directly to the database,
3. a user made changes with medic-conf but neglected to commit their changes, or
4. a user forgets to pull updates from the repo before making their changes.

In any of these cases the configuration will be overwritten.

Instead medic-conf should warn when the configuration is different from when it was last executed. The user should be given the option to overwrite the changes (`force`) or update their configuration to be in sync (eg: `git stash`, export configuration, `git stash pop`).

**Comment by SCdF** [comment]
How would this work? Since presumably *every* time you update a config you're making changes (otherwise why are you updating it!).

Are you thinking we'd establish some kind of "authentic" changes line based on what is in git? I'm struggling to imagine how to solve this cleanly.

One way I thought of would be to store the last pushed `_rev` somewhere in git, presumably in a dotfile in the project directory. This way if you do a pull and the `_rev` is different to what is stored in git you can say that someone has changed. This would work for 1, 2 and 4, but maybe wouldn't work for 3, depending on the perspective I'm supposed to read that from.

So the flow would be:
 - `git pull` to get the latest local medic-project changesz
 - make your changes to the config file
 - `medic-conf whatever`, which checks your .medic-conf file in the project, pulls whatever is in PROD and makes sure the rev is the same
 - If it's bad, shows a diff, and gives you the option of writing out the config to a `config.current.json` file? Which is also in `.gitignore` so people don't think it's useful to commit it?
 - If it's all good, uploads your new config, and then uses the returning `_rev` to update `.medic-conf`, and.... doesn't git commit it? Leaves that to you?

Is that your line of thinking @garethbowen?

**Comment by garethbowen** [comment]
Yeah that's exactly what I was thinking, I just don't like to put any implementation details in issues until they are ready to be worked on.

You would also need to store revs of forms and resources and so on.

I think it would work for number 3 because it would protect you from overwriting and prompt you to export it locally before making your changes and uploading again. You'd lose who made the original changes and why but at least you wouldn't lose the changes!

**Comment by garethbowen** [comment]
Related: #1616
